CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs

```diff
@@ -31,12 +34,86 @@ public StartupLanguageSelect()
             }
         }
 
-        private void ComboBox_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        private string LanguageFallback(string tag)
         {
-            string LangID = LanguageIDIndex[(sender as ComboBox).SelectedIndex];
-            SetAppConfigValue("AppLanguage", LangID);
-            LoadLocale(LangID);
-            NextBtn.IsEnabled = true;
+            tag = tag.ToLower();
+            // English
+            if (tag == "en" || tag.StartsWith("en-")) return "en-us";
+            // Traditional Chinese
+            if (tag == "zh-hant" || tag == "zh-hk" || tag == "zh-mo" || tag == "zh-tw") return "zh-tw";
+            // Chinese defaults to zh-cn
+            if (tag == "zh" || tag.StartsWith("zh-")) return "zh-cn";
+            // German
+            if (tag == "de" || tag.StartsWith("de-")) return "de-de";
+            // Spanish
+            if (tag == "es" || tag.StartsWith("es-")) return "es-419";
+            // Indonesian
+            if (tag == "id" || tag.StartsWith("id-")) return "id-id";
+            // Japanese
+            if (tag == "ja" || tag.StartsWith("ja-")) return "ja-jp";
+            // Korean
+            if (tag == "ko" || tag.StartsWith("ko-")) return "ko-kr";
+            // Polish
+            if (tag == "pl" || tag.StartsWith("pl-")) return "pl-pl";
+            // Portuguese, Portugal
+            if (tag == "pt-pt") return "pt-pt";
+            // Portuguese defaults to pt-br
+            if (tag == "pt" || tag.StartsWith("pt-")) return "pt-br";
+            // Russian
+            if (tag == "ru" || tag.StartsWith("ru-")) return "ru-ru";
+            // Thai
+            if (tag == "th" || tag.StartsWith("th-")) return "th-th";
+            // Vietnamese
+            if (tag == "vi" || tag.StartsWith("vi-")) return "vi-vn";
+            // Default
+            return "en-us";
+        }
+
+        private void RefreshSelection()
+        {
+            SelectLang.SelectedIndex = -1;
+            SelectLang.SelectedIndex = SelectedLangIndex;
+            SelectWindowSize.SelectedIndex = -1;
+            SelectWindowSize.SelectedIndex = SelectedWindowSizeProfile;
+            SelectCDN.UpdateLayout();
```

Comment:
I would personally add a `LogWriteLine`so that we know that the view was refreshed.